src/tiled/main.cpp

```diff
@@ -64,17 +64,30 @@ class CommandLineHandler : public CommandLineParser
 public:
     CommandLineHandler();
 
+    const char *errorString() const { return mError; }
+
+    /**
+     * Used during file export, attempt to determine the output file format
+     * from the command line parameters.
+     * Query errorString if result is null.
+     */
+    template <typename T>
+    T *findExportFormat(const QString *filter,
+                        const QString &targetFile);
```

Comment:
Since `mError` appears to be the only member variable used by this function, I would make that a `QString *` parameter and take this helper function out of the `CommandLineHandler` class, just making it a `static T *findExportFormat(...)`.